test/tstDistributedTree.cpp

```diff
@@ -31,6 +31,41 @@ namespace tt = boost::test_tools;
 
 using ArborX::PairIndexRank;
 
+struct PairRankIndex
+{
+  int rank;
+  int index;
+
+  friend bool operator==(PairRankIndex lhs, PairRankIndex rhs)
+  {
+    return lhs.index == rhs.index && lhs.rank == rhs.rank;
+  }
+  friend bool operator<(PairRankIndex lhs, PairRankIndex rhs)
+  {
+    return lhs.rank < rhs.rank ||
+           (lhs.rank == rhs.rank && lhs.index < rhs.index);
+  }
+  friend std::ostream &operator<<(std::ostream &stream,
+                                  PairRankIndex const &pair)
+  {
+    return stream << '[' << pair.rank << ',' << pair.index << ']';
+  }
+};
+
+struct DistributedNearestCallback
+{
+  using tag = ArborX::Details::ConstrainedNearestCallbackTag;
+
+  int rank;
+
+  template <typename Predicate, typename Value, typename OutputFunctor>
+  KOKKOS_FUNCTION void operator()(Predicate const &, Value const &value,
+                                  OutputFunctor const &out) const
+  {
+    out({rank, value});
```

Comment:
Why do we need both this one and `DefaultCallbackWithRank`?
One aggregates the rank first and the value second, the other swaps the two.

Reply:
We don't *need* it. Rather, this is just a custom callback that does this.

Reply:
Maybe add a comment to that effect then.

Reply:
Done.